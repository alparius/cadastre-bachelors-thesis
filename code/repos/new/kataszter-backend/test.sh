#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
