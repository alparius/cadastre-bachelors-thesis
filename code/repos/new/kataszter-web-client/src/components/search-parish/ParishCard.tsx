import React from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { NavLink } from "react-router-dom";
import { Card, Header, Icon, Popup } from "semantic-ui-react";

import IParishMin from "../../data/parish/ParishMin";
import ItalicFM from "../../util/styledMessages/ItalicFM";

interface IParishCardProps {
    parish: IParishMin;
}

const ParishCard: React.FC<IParishCardProps> = (props: IParishCardProps) => {
    const { parish } = props;
    const { formatMessage: fm } = useIntl();

    return (
        <Card key={parish.ID} fluid as={NavLink} to={`/parishes/profile/${parish.ID}`}>
            <Card.Content>
                <Card.Header>
                    <Header as="h2">
                        {parish.name}
                        <span style={{ float: "right" }}>
                            {parish.verified ? (
                                <Popup
                                    content={fm({ id: "profile.verified.isVerifiedPopup" })}
                                    trigger={<Icon name="check circle outline" fitted color="green" />}
                                />
                            ) : (
                                <Popup
                                    content={fm({ id: "profile.verified.notVerifiedPopup" })}
                                    trigger={<Icon name="question circle outline" fitted color="grey" />}
                                />
                            )}
                        </span>
                    </Header>
                </Card.Header>
                <Card.Description>
                    <Card.Content>
                        <FM id="label.numberOfPriests" />: {parish.numberOfPriests ? parish.numberOfPriests : <ItalicFM id="general.noData" />}
                        <br />
                        <FM id="label.parish.cityName" />: {parish.cityName ? parish.cityName : <ItalicFM id="general.noData" />}
                    </Card.Content>
                </Card.Description>
            </Card.Content>
        </Card>
    );
};

export default ParishCard;
