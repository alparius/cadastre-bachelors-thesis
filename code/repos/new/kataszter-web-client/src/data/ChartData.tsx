interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
