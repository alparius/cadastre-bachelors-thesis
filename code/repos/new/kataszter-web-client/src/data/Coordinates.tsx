interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
