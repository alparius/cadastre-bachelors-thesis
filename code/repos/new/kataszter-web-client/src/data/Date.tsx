interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
