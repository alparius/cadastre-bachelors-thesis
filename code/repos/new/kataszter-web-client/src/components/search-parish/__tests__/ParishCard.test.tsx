/*===================
  PriestCard.unit.test.tsx
 ===================*/
import React from "react";

import IParishMin from "../../../data/parish/ParishMin";
import { NewTestParishMin } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../util/test/test-wrapper";
import ParishCard from "../ParishCard";

describe("ParishCard element", () => {
    it("should render a card with the parish data", () => {
        const regularParish: IParishMin = NewTestParishMin;
        const container = createComponentWithIntlAndRouter(<ParishCard parish={regularParish} />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
