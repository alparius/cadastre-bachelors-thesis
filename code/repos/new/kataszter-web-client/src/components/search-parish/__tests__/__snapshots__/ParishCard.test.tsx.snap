// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParishCard element should render a card with the parish data 1`] = `
<a
  aria-current={null}
  className="ui fluid card"
  href="/parishes/profile/1"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <h2
        className="ui header"
      >
        name
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <i
            aria-hidden="true"
            className="green check circle outline fitted icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </span>
      </h2>
    </div>
    <div
      className="description"
    >
      <div
        className="content"
      >
        Number of priests
        : 
        12
        <br />
        Town
        : 
        Test city
      </div>
    </div>
  </div>
</a>
`;
